docs/building-images.md

```diff
@@ -234,6 +273,24 @@ There are a number of optional directories that may be included in the image con
 
 The following sections further describe optional directories that may be included.
 
+### Kubernetes
+
+* `kubernetes` - May be included to inject cluster specific configurations, apply manifests and install Helm charts.
+  * `config` - Contains [k3s](https://docs.k3s.io/installation/configuration#configuration-file) or
+    [RKE2](https://docs.rke2.io/install/configuration#configuration-file) cluster configuration files
+    * `server.yaml` - If present, this configuration file will be applied to all control plane nodes
+    * `agent.yaml` - If present, this configuration file will be applied to all worker nodes
+  * `manifests` - Contains locally provided manifests which will be applied to the cluster. All files in this directory
```

Comment:
Maybe a note that this can be used in combination (or as a replacement) for the `kubernetes.manifests` in image definition?